### PR TITLE
Create round-system

### PR DESCRIPTION
**Pull request by PenguinSatan**

### CODE REVIEW

File: battle/round-system.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  for (let i = 0; i < team.length; i += 1) {
+    team2.push(unitToTurnLogItem(team[i]));
+  }
+  return team2;
+}; */
+
+const turnLog = [...alliedParty.map(copyUnit), ...enemyParty.map(copyUnit)];
+
+const sortUnitByMovepoints = (turnLogToSort) => turnLogToSort.sort(
+  (movePoints1, movePoints2) => movePoints2.movePoints - movePoints1.movePoints,
```
Comment (kujiraOo) [inline battle/round-system.js:74]: rename `movePoints1, movePoints2` to either `unit1, unit2` or just `a, b`

File: battle/round-system.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+const pickUnitWithMaxMovepoints = (units) => {
+  sortUnitByMovepoints(units);
+  return units[0];
+};
+
+const moveCost = 6;
+
+const calculateMovepoints = (units, movingUnitId) => {
+  const newUnitsArray = units.map(copyUnit);
```
Comment (kujiraOo) [inline battle/round-system.js:85]: i would prefer `newUnitsArray` -> `newUnitArray`

File: battle/round-system.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+  movePoints: 10,
+}];
+
+const unitToTurnLogItem = ({
+  id, name, agl, movePoints,
+}) => ({
+  id, name, agl, movePoints,
+});
+
+const copyUnit = (unit) => ({ ...unit });
```
Comment (kujiraOo) [inline battle/round-system.js:59]: maybe rename it to `copyObject`, as it actually works with any object, not only `unit`

**Review by kujiraOo — CHANGES_REQUESTED** [review]

**Comment by kujiraOo** [comment]
closes #22

**Comment by kujiraOo** [comment]
please rename the PR title to smth like "Add logic for calculating turn queue"

File: battle/create-unit.js [hunk 4]
```diff
@@ -1,13 +1,17 @@
...
-  id,
-  name,
-  hp: random.randomHp(),
-  mp: random.randomMp(),
-  agl: random.randomAgl(),
-  dmg: random.randomDmg(),
-  def: random.randomDef(),
-});
+const createUnit = (id, name) => {
+  const aglRng = random.randomAgl();
```
[the remark this answers is not in the document]
Comment (PenguinSatan) [reply battle/create-unit.js:4]: done

File: battle/round-system.js [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+  }
+  return queue;
+};
+// console.log(sortUnitByMovepoints(turnLog).map(unitToTurnLogItem));
+
+// console.log(calculateMovepoints(turnLog, 2).map(unitToTurnLogItem));
+
+console.log(turnQueue(turnLog).map(unitToTurnLogItem));
+
+// console.log(restoreMovepoints(turnLog));
```
[the remark this answers is not in the document]
Comment (PenguinSatan) [reply battle/round-system.js:130]: fixed, unused things moved to coments for future use

File: battle/round-system.js [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+    queue.push(unit);
+    tempUnits = calculateMovepoints(tempUnits, unit.id);
+  }
+  return queue;
+};
+// console.log(sortUnitByMovepoints(turnLog).map(unitToTurnLogItem));
+
+// console.log(calculateMovepoints(turnLog, 2).map(unitToTurnLogItem));
+
+console.log(turnQueue(turnLog).map(unitToTurnLogItem));
```
[the remark this answers is not in the document]
Comment (PenguinSatan) [reply battle/round-system.js:128]: 126 deleted

File: battle/round-system.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (PenguinSatan) [reply battle/round-system.js:130]: done

File: battle/round-system.js (hunk 2 above)
Comment (PenguinSatan) [reply battle/round-system.js:85]: as u wish

File: battle/round-system.js (hunk 3 above)
Comment (PenguinSatan) [reply battle/round-system.js:59]: agree

**Review by kujiraOo — APPROVED** [review]

**Pull request merged**